src/verifier/builder_visitor.rs

```diff
@@ -254,10 +250,20 @@ impl SymbolTableBuilder<'_> {
                         Some(other_loc.line_col.clone()),
                     );
                 } else {
-                    // This should never be the case since it's controlled by the compiler!
-                    self.err
-                        .unexpected_error(true, Some(UNEXPECTED_ERR_MSG.to_string()), None);
-                    unreachable!()
+                    // If there is another fn with the same name as a compiler generated fn, throw a duplicate id error
+                    match &f_name.loc {
+                        Some(loc) => {
+                            self.err.compiler_fn_overload_error(
+                                false,
+                                f_name.name.clone(),
+                                Some(loc.line_col.clone()),
+                            );
+                        }
+                        None => {
+                            self.err
+                                .compiler_fn_overload_error(false, f_name.name.clone(), None);
+                        }
+                    }
```

Comment:
Isn't there a `is_comp_provided` field that we could use here? If `is_comp_provided` is `true`, then we'd need this `compiler_fn_overload_error`. Otherwise, it's truly an issue with the compiler not populating the `loc` for some script content.

Reply:
Can you explain what you mean by this when you get a chance?

Reply:
we dont have a "is-comp-provided" for FnId, so instead of changing that I used that user def fn all have a loc where compiler-def ones dont. If you'd like me to change that, I can though